### PR TITLE
[Tradeskill] Fix logic in taught tradeskill recipes

### PR DESCRIPTION
**Pull request by Natedog2012**

### CODE REVIEW

File: zone/tradeskills.cpp [hunk 1]
```diff
@@ -810,6 +810,14 @@ void Client::SendTradeskillSearchResults(
 				continue;
 			}
 		}
+		
+		//Check if we need to learn it before sending them the recipe.. 
+		DBTradeskillRecipe_Struct spec;
+		if (content_db.GetTradeRecipe(recipe_id, objtype, someid, this->CharacterID(), &spec)) {
```
Comment (Akkadius) [inline zone/tradeskills.cpp:816]: Doing a select query inside of a loop like this is not ideal. We should fetch recipes once and then loop through it in memory to perform this check

Comment (Natedog2012) [reply zone/tradeskills.cpp:816]: Righto I see a better way to fix it.. but looking at it.. I see other issues

**Review by Kinglykrab — APPROVED** [review]
Looks good to me. Can’t believe we never noticed that issue with row data.

**Review by Akkadius — APPROVED** [review]

**Comment by Natedog2012** [comment]
> e

Makes me wonder if those checks... row[5] == nullptr even work :D

**Pull request merged**